chimney/src/main/scala/io/scalaland/chimney/TransformerFSupport.scala

```diff
@@ -0,0 +1,97 @@
+package io.scalaland.chimney
+
+import scala.collection.compat._
+import scala.language.higherKinds
+
+trait TransformerFSupport[F[_]] {
+
+  def pure[A](value: A): F[A]
+
+  def combine[A, B](fa: F[A], fb: => F[B]): F[(A, B)]
+
+  def map[A, B](fa: F[A], f: A => B): F[B]
+
+  def traverse[M[X] <: Iterable[X], A, B](mfa: M[A], f: A => F[B])(
```

Comment:
Iterable upper bound restricts users to use collections which implement this interface.
It makes impossible to use third-party collections like `cats.data.Chain`.
Isn't it? If you use type class as a constraint, I think it will be much better.

What do you think?

Reply:
Let's make a distinction between `F[_]` and `M[_]` here.

`traverse` is invoked for every field of type `Seq[A]` that needs to be transformed into `Seq[B]`, but we have (or can derive) `TransformerF` between `A` and `B`, so a function of type `A => F[B]`; `traverse` allows us to lift the function on a collection level, so we can obtain `Seq[A] => F[Seq[B]]`. In user's code we will not have only `Seq`s, but probably other types. The minimal interface that we need in order to provide reasonable implementation of `traverse` is `IterableOnce`.

I can guess why you are worried about `cats.data.Chain` (correct me if I got it bad) - you would like to capture transformation errors into `cats.data.Chain`, for example: in `ValidatedNec[T]` (which Chain, or NonEmptyChain is a part of) which is an instantiation of `F[_]`, not `M[_]`.

`M[_]` here expresses all the types that automatic conversions are supported by Chimney out of the box (which is Iterable in fact). If you want to support any other `M[_]` which is not `Iterable`, which may happed if you use `cats.data.Chain` to model your domain, you just need to provide  implicit instance of `TransformerF[F[_], M[A], M[B]]`, so actually: `M[A] => F[M[B]]`, probably given an instance of `TransformerF[F[_], A, B]`, so `A => F[B]`.

If you want, I may provide a test case that illustrates it.

Reply:
You are right. I confused with `F[_]`